src/MSBuild/XMake.cs

```diff
@@ -828,6 +873,32 @@ private static void DebuggerLaunchCheck()
 
                     string timerOutputFilename = Environment.GetEnvironmentVariable("MSBUILDTIMEROUTPUTS");
 
+                    if ((getProperty.Length > 0 || getItem.Length > 0 || getTargetResult.Length > 0) && targets?.Length > 0 && result is not null)
+                    {
+                        ProjectInstance builtProject = result.ProjectStateAfterBuild;
+
+                        if (builtProject is null)
+                        {
+                            // Build failed; do not proceed
+                            ILogger simpleLogger = loggers.FirstOrDefault(l => l is SimpleErrorLogger);
+                            string errorMessage = simpleLogger is null ? "internal error" : (simpleLogger as SimpleErrorLogger).errorList.ToString();
+                            Console.Error.WriteLine(ResourceUtilities.FormatResourceStringStripCodeAndKeyword("BuildFailedWithPropertiesItemsOrTargetResultsRequested", errorMessage));
```

Comment:
I don't think the `builtProject is null` condition is sufficient: I am not seeing this.

```xml
<Project>
    <Target Name="Go">
        <Error Text="Badness happened" />
    </Target>
</Project>
```

```sh-session
❯ .\.dotnet\dotnet.exe msbuild .\bar.proj -getTargetResult:Go
{
  "Target Results": {
    "Go": {
      "Result": "Failure",
      "Items": []
    }
  }
}
```

Reply:
This needs a regression test.

Reply:
I think I see the error condition here as more specific than you do. I see the error case as "something catastrophic happened, and we can't give you what you want." You see it as "the build failed for some reason." I included the target's success/failure specifically because I'd consider "Failure" as a valid output. If you care about whether it succeeded or failed, you can check for that; if not, you may get more information by skipping past the error message and trying to provide what the user requested.

Reply:
Others can weigh in, but I see this feature as pointed towards "I want more information about my build" more than "my build failed, and I want to know why." If your build failed, just collect a binlog; it'll be a lot more helpful than anything I could give you here. (And you can still collect a binlog with this feature enabled.)

Reply:
Suppose you have a build definition where you have a script that does a design-time-like build to get information about the project output. In CI, long after this is established, it starts failing. What is the user experience you expect in this case? What does the scripting look like to handle failures?

Reply:
I expect the user to look at the binlog that is also enabled in their CI and use that to diagnose the failure.

Reply:
I'd really like an answer to this part

> What does the scripting look like to handle failures?

Reply:
I don't think I understand that question. What kind of failure? Like do you mean something like the `<Error />` you mentioned earlier or something more serious? And are you asking how the user is supposed to diagnose the problem, how this feature should handle it, or something else?

Reply:
@Forgind when I was thinking "errors should go to stderr" what I was imagining was build errors/warnings (in addition to any catastrophic errors/crashes/etc). given @rainersigwald's sample above, I'd expect to see something like:

```shell
❯ .\.dotnet\dotnet.exe msbuild .\bar.proj -getTargetResult:Go
.\bar.proj(3:8) - error: Badness happened
{
  "Target Results": {
    "Go": {
      "Result": "Failure",
      "Items": []
    }
  }
}
```

with the diagnostic being emitted on stderr, and the json structure emitted to stdout. This is because in shell scripting it's easiest to manage those streams separately (think streaming stdout into `jq` for example) if you don't also have to preprocess out any error lines.

Reply:
So to be clear, your proposal is that all errors MSBuild emits would go straight to stderr, but we'd generally continue on (if possible) and emit target results as it currently does?

Reply:
@Forgind yes, with @rainersigwald's exit code caveat :+1:

Reply:
I tested with the changes up to `cad93f2` and the exit code is done, but the emitting of msbuild diagnostics to stderr isn't implemented yet @Forgind